File: app/internal/store/log_repo.go
package store

// 数据层：操作日志(logs)（均为 *Store 方法，满足 LogRepository）。

import (
	"time"

	"heritage-mgmt/internal/domain"
)

func (s *Store) InsertLog(action, target, detail string) {
	if s == nil || s.db == nil {
		return
	}
	s.db.Exec("INSERT INTO logs(ts,action,target,detail) VALUES(?,?,?,?)",
		time.Now().Format("2006-01-02 15:04:05"), action, target, detail)
}

func (s *Store) ListLogs(limit int) ([]domain.LogEntry, error) {
	rows, err := s.db.Query("SELECT id,ts,action,target,detail FROM logs ORDER BY id DESC LIMIT ?", limit)
	if err != nil {
		return nil, err
	}
	defer rows.Close()
	var out []domain.LogEntry
	for rows.Next() {
		var l domain.LogEntry
		rows.Scan(&l.ID, &l.Ts, &l.Action, &l.Target, &l.Detail)
		out = append(out, l)
	}
	return out, nil
}

File: desktop/src/core/storage/LogRepo.cpp
#include "LogRepo.h"

#include <QDateTime>
#include <QSqlQuery>

namespace heritage {

LogRepo::LogRepo(QSqlDatabase db) : db_(db) {}

void LogRepo::insert(const QString& action, const QString& target, const QString& detail) {
    QSqlQuery q(db_);
    if (!q.prepare(QStringLiteral("INSERT INTO logs(ts,action,target,detail) VALUES(?,?,?,?)")))
        return;
    q.addBindValue(QDateTime::currentDateTime().toString(QStringLiteral("yyyy-MM-dd HH:mm:ss")));
    q.addBindValue(action);
    q.addBindValue(target);
    q.addBindValue(detail);
    q.exec();
}

QVector<LogEntry> LogRepo::list(int limit) {
    QVector<LogEntry> out;
    QSqlQuery q(db_);
    if (!q.prepare(QStringLiteral("SELECT id,ts,action,target,detail FROM logs ORDER BY id DESC LIMIT ?")))
        return out;
    q.addBindValue(limit);
    if (!q.exec())
        return out;
    while (q.next()) {
        LogEntry l;
        l.id = q.value(0).toLongLong();
        l.ts = q.value(1).toString();
        l.action = q.value(2).toString();
        l.target = q.value(3).toString();
        l.detail = q.value(4).toString();
        out.append(l);
    }
    return out;
}

} // namespace heritage
